influxql/functions.gen.go

```diff
@@ -377,6 +380,50 @@ func (r *FloatElapsedReducer) Emit() []IntegerPoint {
 	return nil
 }
 
+// FloatSampleReduces implements a reservoir sampling to calculate a random subset of points
+type FloatSampleReducer struct {
+	count int // how many points we've iterated over
+
+	points floatPoints // the reservoir
+}
+
+// NewFloatSampleReducer creates a new FloatSampleReducer
+func NewFloatSampleReducer(size int) *FloatSampleReducer {
+	return &FloatSampleReducer{
+		points: make(floatPoints, size),
+	}
+}
+
+// AggregateFloat aggregates a point into the reducer.
+func (r *FloatSampleReducer) AggregateFloat(p *FloatPoint) {
+	// Fill the reservoir with the first n points
+	if r.count < len(r.points) {
+		r.points[r.count] = *p
+		r.count++
+		return
+	}
+
+	// Generate a random integer between 1 and the count and
+	// if that number is less than the length of the slice
+	// replace the point at that index rnd with p.
+	rnd := rand.Intn(r.count + 1)
```

Comment:
Along those lines I think its mostly correct as is. 
See the second answer [here](http://math.stackexchange.com/questions/890272/how-to-pick-a-random-sample-of-given-size-from-a-set-of-unknown-size).

It provides a nice algorithm for selecting a sample from an unknown sized set.

I saw two differences:

1. If the total number of points selected is less than size, it is technically not a valid sample. This may not matter in this case but we should discuss.
2. It seems that rand value should be `prob := size/count` where count is the number of points seen. Then with probability `prob` evict a random value from the list and replace it with the current point `p`.  

So to avoid the division and double reads of a random value one could rearrange the expression from:

```
prob := size / count
if rand.Float() < prod {
    i := rand.Intn(r.count+1)
    r.points[i] = *p
}
```

to

```
i := rand.Intn(count +1)
if i < size {
    r.points[i] = *p
}
```

Seems like it is correct to me except r.count needs to be incremented for all new points not just the first size points.
With which we arrive at @desa's solution, I am not sure that reusing the rand Intn is ok but it seems like it should be fine.